nomos-core/src/block.rs

```diff
@@ -38,34 +44,41 @@ impl<TxId: Clone + Eq + Hash + Serialize + DeserializeOwned> Block<TxId> {
                 leader_id: proposer,
             },
         };
-
         let mut s = Self {
             header,
-            transactions,
             beacon,
+            transactions,
+            blobs,
         };
         let id = block_id_from_wire_content(&s);
         s.header.id = id;
         s
     }
 }
 
-impl<TxId: Clone + Eq + Hash> Block<TxId> {
+impl<Tx: Clone + Eq + Hash, Blob: Clone + Eq + Hash> Block<Tx, Blob> {
     pub fn header(&self) -> &consensus_engine::Block {
         &self.header
     }
 
-    pub fn transactions(&self) -> impl Iterator<Item = &TxId> + '_ {
+    pub fn transactions(&self) -> impl Iterator<Item = &Tx> + '_ {
         self.transactions.iter()
     }
 
+    pub fn blobs(&self) -> impl Iterator<Item = &Blob> + '_ {
```

Comment:
isn't a blob just a type of transaction? or do we mean cl transactions and data blobs?

Reply:
yes, transactions are `cl`transactions. Blobs just data blobs. I didn't wanted to include the distinction in the naming. But it may be worth it. `cl_transactions` and `bl_blobs` as Alvaro suggested?

Reply:
> cl_transactions and bl_blobs as Alvaro suggested?

This might be clearer, at least until we have something visible for CL and EZ in the future.